x-pack/legacy/plugins/siem/public/pages/case/components/add_comment/index.test.tsx

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import { mount } from 'enzyme';
+
+import { AddComment } from './';
+import { TestProviders } from '../../../../mock';
+import { getFormMock } from '../__mock__/form';
+import { Router, routeData, mockHistory, mockLocation } from '../__mock__/router';
+
+import { useInsertTimeline } from '../../../../components/timeline/insert_timeline_popover/use_insert_timeline';
+import { usePostComment } from '../../../../containers/case/use_post_comment';
+jest.mock('../../../../components/timeline/insert_timeline_popover/use_insert_timeline');
+jest.mock('../../../../containers/case/use_post_comment');
+import { useForm } from '../../../../../../../../../src/plugins/es_ui_shared/static/forms/hook_form_lib/hooks/use_form';
+import { wait } from '../../../../lib/helpers';
+jest.mock(
+  '../../../../../../../../../src/plugins/es_ui_shared/static/forms/hook_form_lib/hooks/use_form'
+);
+
+export const useFormMock = useForm as jest.Mock;
+
+const useInsertTimelineMock = useInsertTimeline as jest.Mock;
+const usePostCommentMock = usePostComment as jest.Mock;
+
+const onCommentSaving = jest.fn();
+const onCommentPosted = jest.fn();
+const postComment = jest.fn();
+const handleCursorChange = jest.fn();
+const handleOnTimelineChange = jest.fn();
+
+const addCommentProps = {
+  caseId: '1234',
+  disabled: false,
+  insertQuote: null,
+  onCommentSaving,
+  onCommentPosted,
+  showLoading: false,
+};
+
+const defaultInsertTimeline = {
+  cursorPosition: {
+    start: 0,
+    end: 0,
+  },
+  handleCursorChange,
+  handleOnTimelineChange,
+};
+
+const defaultPostCommment = {
+  isLoading: false,
+  isError: false,
+  postComment,
+};
+const sampleData = {
+  comment: 'what a cool comment',
+};
+describe('AddComment ', () => {
+  const formHookMock = getFormMock(sampleData);
+
+  beforeEach(() => {
+    jest.resetAllMocks();
+    useInsertTimelineMock.mockImplementation(() => defaultInsertTimeline);
+    usePostCommentMock.mockImplementation(() => defaultPostCommment);
+    useFormMock.mockImplementation(() => ({ form: formHookMock }));
+    jest.spyOn(routeData, 'useLocation').mockReturnValue(mockLocation);
+  });
+
```

Comment:
I think we should add a test that checks the existence (initial render) of the component.

Reply:
initial render is proved by sub component render checks. ill add a disabled check